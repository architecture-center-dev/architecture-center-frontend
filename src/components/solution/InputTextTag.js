import React, { useEffect, useRef } from 'react';
import PropTypes from "prop-types";
import { TextField } from '@material-ui/core';

const InputTextTag = ({label, onKeyUp, onFocusOut}) => {
    const inputEl = useRef(null);
    useEffect(() => {
        inputEl.current.querySelector('input').focus();
        
    },[])
    return(
        <TextField 
            ref={inputEl}
            label={label} 
            variant="outlined" 
            fullWidth
            onKeyUp={onKeyUp}
            onBlur={onFocusOut}
            size="small"
            />
    )
}

export default InputTextTag;

InputTextTag.prototype = {
    label: PropTypes.string.isRequired,
    onKeyUp: PropTypes.func.isRequired,
    onFocusOut: PropTypes.func.isRequired,
}