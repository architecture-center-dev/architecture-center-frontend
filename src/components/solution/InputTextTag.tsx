import React, { useEffect, useRef } from 'react';
import { TextField } from '@material-ui/core';

export type InputTextTagProps = {
    label: string, 
    onKeyUp: any, 
    onFocusOut: any
}

const InputTextTag = ({label, onKeyUp, onFocusOut}: InputTextTagProps) => {
    const inputEl: any = useRef(null);
    useEffect(() => {
        inputEl.current.querySelector('input').focus();
        
    },[])
    return(
        <TextField 
            data-testid="input-tag"
            ref={inputEl}
            label={label} 
            variant="outlined" 
            fullWidth
            onKeyUp={onKeyUp}
            onBlur={onFocusOut}
            size="small"
            />
    )
}

export default InputTextTag;
