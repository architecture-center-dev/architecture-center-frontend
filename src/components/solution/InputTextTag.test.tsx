import * as React from "react";
import InputTextTag, { InputTextTagProps } from "./InputTextTag";
import { render, screen, fireEvent } from '@testing-library/react'

describe("<InputTextTag />", () => {
    it("Component should render correctely", async () => {

        const props: InputTextTagProps = {
            onKeyUp: jest.fn(),
            onFocusOut: jest.fn(),
            label: "test label"
        }

        render(<InputTextTag {...props} />);
        const textField = screen.getByTestId('input-tag');

        expect(textField).toBeTruthy();
        expect(textField).toMatchSnapshot();
    });

    it("Press some key should be call a event", async () => {

        const props: InputTextTagProps = {
            onKeyUp: jest.fn(),
            onFocusOut: jest.fn(),
            label: "test label"
        }

        render(<InputTextTag {...props} />);
        const textField = screen.getByTestId('input-tag');

        fireEvent.keyUp(textField,{ value: "some value" });

        expect(props.onKeyUp).toBeCalled();
        expect(textField).toMatchSnapshot();
    });
});
