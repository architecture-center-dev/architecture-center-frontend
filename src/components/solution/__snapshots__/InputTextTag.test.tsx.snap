// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputTextTag /> Component should render correctely 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  data-testid="input-tag"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-focused Mui-focused"
    data-shrink="true"
  >
    test label
  </label>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
      type="text"
      value=""
    />
    <fieldset
      aria-hidden="true"
      class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
      >
        <span>
          test label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`<InputTextTag /> Press some key should be call a event 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  data-testid="input-tag"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-focused Mui-focused"
    data-shrink="true"
  >
    test label
  </label>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
      type="text"
      value=""
    />
    <fieldset
      aria-hidden="true"
      class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
      >
        <span>
          test label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
